### ISSUE TITLE
lscount returns empty set instead of 0 if no documents are found

### ISSUE THREAD

**Issue by lukaspustina** [bosun-monitor/bosun#1160]
I'm using bosun in version 0.2.0 and the following query to check for grok parse failures:
```
lscount("logstash", "domain", "tags:_grokparsefailure", "5m", "5m", "") * 600
```
This works fine as long as _there are_ documents with the specific tag. But if not, the result seems to be an empty set instead of 0. As a consequence, any alarm on this query never get back to normal / green.

**Comment by kylebrandt** [comment]
Could also be related since there is also state issues: https://github.com/bosun-monitor/bosun/issues/846

I'm not sure what could be returned here since Bosun doesn't know what the sets for `domain` would be to set to zero... :/

**Comment by lukaspustina** [comment]
I get the same result even when leaving `keyString` (`""`) empty.

This used to work when I was compiling Bosun myself. Since 0.2.0 I'm using your releases. Same behavior in 0.3.0. Maybe that helps do find the culprit.

**Comment by kylebrandt** [comment]
I think what has to happen to fix this correctly is have a different state workflow for this. So basically unknown for this needs to = "normal". The implementation of that scares me.

Another possibility that just occurred to me would be to issue double queries, one with the filterstring and one without (behind the scenes). Then diff the two. That might be more sane - more thought required.

My idea for a partial workaround which I'm going to try to play around with today is using the `log = true` directive. This means it will never appear on the dashboard though :-/.

**Comment by awesensepaul** [comment]
Did you find a solution to this @kylebrandt ? I would prefer the alerts be visible on the dashboard, so I think perhaps the second suggestion (diff two queries) might be the easiest to accomplish this.

There is the possibility that no log messages occur at all during a check interval, but in practice I don't think that would ever occur (at least for our logstash setup).

**Comment by awesensepaul** [comment]
I thought about this, from the point-of-view of my team's preferred workflow. For us, it is important that any incident (even log-based) has a lifecycle on the bosun dashboard.

I think the closest way to do this is to use `ignoreUnknown = true`. The alert stays critical (or warning), until it has been acknowledged, even if that is longer than a check interval. 

What is missing, is a way for alerts like this to be "closed" or "forgotten" even though they are active.

**Comment by deanefrati** [comment]
@kylebrandt what about adding a is null/empty is not null/empty operator? In that case we can set our alert to use is not null for a crit/warn and then have it set to normal if the result is empty. That way we can still use the alert lifecycle in the dashboard and don't need to change the behavior of normal or do fancy stuff with query comparison.

**Comment by kylebrandt** [comment]
I think maybe just ignoreUknown needs to make unknown go back to normal in this case. That way the alert will no longer be active, and should be able to be closed.

**Comment by kylebrandt** [comment]
Shoud be fixed in #1620
